src/metabase/models/permissions.clj

```diff
@@ -573,15 +572,10 @@
 ;; somewhat useful?
 (defn- check-native-and-schemas-permissions-allowed-together [{:keys [native schemas]}]
   ;; Only do the check when we have both, e.g. when the entire graph is coming in
-  (if-not (and native schemas)
-    :ok
-    (match [native schemas]
-      [:write :all]  :ok
-      [:write _]      (log/warn "Invalid DB permissions: if you have write access for native queries, you must have full data access.")
-      [:read  :none]  (log/warn "Invalid DB permissions: if you have readonly native query access, you must also have data access.")
-      ;; TODO -- log a warning if we end up keeping this code.
-      [_      :block] false
-      [_      _]     :ok)))
+  (if (and (= native :write)
+           (not= schemas :all))
+    (log/warn (trs "Invalid DB permissions: If you have write access for native queries, you must have full data access."))
```

Comment:
i'd do an explicit `(do (log/warn "...") nil)` so its incredibly clear that we intend to return nil from this function on the then branch here. I had to parse it a few times while reading thinking there was a missing branch before i realized the nil return of the log was functional and intentional herel

Reply:
👍 good idea